app/assets/javascripts/catarse_pagarme/credit_card.js

```diff
@@ -104,16 +62,64 @@ App.views.Pagarme.addChild('PaymentCard', _.extend({
     $(e.currentTarget).hide();
     that.parent.loader.show();
 
+    if( that.hasSelectedSomeCard() ) {
+      that.requestPayment({
+        card_id: this.selectedCard().val(),
+        payment_card_installments: that.getInstallments()
+      });
+    } else {
+      PagarMe.encryption_key = this.$('.pagarme-e-key').data('key');
+
+      var creditCard = new PagarMe.creditCard();
```

Comment:
I would break the following code in several smaller methos to improve readability to have somethng like this:
```javascript
var creditCard = this.newCreditCard();
if(this.cardHasErrors(creditCard))
{
  this.displayErrors(creditCard);
}
```